### PR TITLE
Add poisson-boltzmann to the SCRF

### PR DESCRIPTION
**Pull request by Gabrielgerez**
Here I have added the alternative to solve the 1:1 Poisson-Boltzmann equation as a part of the SCRF procedure. Right now the parameters are hardcoded but will soon make them user-determined.

### CODE REVIEW

File: python/template.yml [hunk 1]
```diff
@@ -948,6 +949,35 @@ sections:
             docstring: |
               Formulation of the Permittivity function. Currently only the
               exponential is used.
+      - name: D_H_screening
+        docstring: |
+          Parameters for the debye-huckel screening factor
```
Comment (robertodr) [inline python/template.yml:954]:
```suggestion
          Parameters for the Debye-Huckel screening factor
```

File: src/environment/Cavity.h [hunk 2]
```diff
@@ -75,6 +76,12 @@ class Cavity final : public mrcpp::RepresentableFunction<3> {
     std::vector<double> getWidths() const { return this->sigmas; }                //!< Returns #sigmas.
     std::vector<double> getWidthScalings() const { return this->betas; }          //!< Returns #betas.
 
+    void setCoordinates(const std::vector<mrcpp::Coord<3>> &coords) { this->centers = coords; } //!< Sets #centers.
+    void setRadii(const std::vector<double> &R) { this->radii_0 = R; }                          //!< Sets #radii_0.
+    void setRadiiScalings(const std::vector<double> &a) { this->alphas = a; }                   //!< Sets #alphas.
+    void setWidths(const std::vector<double> &s) { this->sigmas = s; }                          //!< Sets #sigmas.
+    void setWidthScalings(const std::vector<double> &b) { this->betas = b; }                    //!< Sets #betas.
+
```
Comment (robertodr) [inline src/environment/Cavity.h:84]: these appear to be unused, so you can remove them.
```suggestion
```

File: src/environment/Cavity.h [hunk 3]
```diff
@@ -66,6 +66,7 @@ class Cavity final : public mrcpp::RepresentableFunction<3> {
 public:
     Cavity(const std::vector<mrcpp::Coord<3>> &coords, const std::vector<double> &R, const std::vector<double> &alphas, const std::vector<double> &betas, const std::vector<double> &sigmas);
     Cavity(const std::vector<mrcpp::Coord<3>> &coords, const std::vector<double> &R, double sigma);
+    Cavity(){};
```
Comment (robertodr) [inline src/environment/Cavity.h:69]: I'm not sure why you need this CTOR, I don't think there's ever a situation where the default CTOR is needed.

File: src/environment/DHScreening.h [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+class DHScreening final : public mrcpp::RepresentableFunction<3> {
+public:
+    /** @brief Standard constructor. Initializes the #cavity_ion and #kappa_out with the input parameters.
+     *  @param cavity_ion interlocking spheres of Cavity class.
+     *  @param kappa_out value of the screening function outside the #cavity_ion.
+     *  @param formulation Decides which formulation of the #DHScreening function to implement, only continuous screening function available
+     * available as of now.
+     */
+    DHScreening(const Cavity cavity_ion, double kappa_out, std::string formulation);
```
Comment (robertodr) [inline src/environment/DHScreening.h:53]: Pass large objects which you're not going to modify by `const` reference https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-in
```suggestion
    DHScreening(const Cavity & cavity_ion, double kappa_out, const std::string & formulation);
```

File: src/environment/Cavity.h (hunk 3 above)
Comment (Gabrielgerez) [reply src/environment/Cavity.h]: This was part of early testing and remained until now, thanks for spotting it :)

**Review by ilfreddy — CHANGES_REQUESTED** [review]

File: src/environment/Cavity.h (hunk 2 above)
Comment (Gabrielgerez) [reply src/environment/Cavity.h]: They are used by the printer methods in the Permittivity and DHScreening classes

File: src/driver.cpp
[the remark this answers is not in the document]
Comment (Gabrielgerez) [reply src/driver.cpp:1]: I think I am following the convention. Class is first uppercase then camelcase. Member methods are simply camelcase and member variables are `_` separated

File: src/environment/GPESolver.h
[the remark this answers is not in the document]
Comment (Gabrielgerez) [reply src/environment/GPESolver.h:1]: I have now added documentation, check that it is acceptable

File: src/qmfunctions/qmfunction_utils.h [hunk 5]
```diff
@@ -28,6 +28,7 @@
 #include "mrchem.h"
 #include "qmfunction_fwd.h"
 
+using mrcpp::ComplexFunctionVector;
```
[the remark this answers is not in the document]
Comment (Gabrielgerez) [reply src/qmfunctions/qmfunction_utils.h:31]: Wasnt necessary

[the remark this answers is not in the document]
Comment (Gabrielgerez) [reply src/qmfunctions/qmfunction_utils.h]: properly removed now

File: src/environment/Cavity.h (hunk 2 above)
Comment (Gabrielgerez) [reply src/environment/Cavity.h]: Correction, it was the getters that were used. These are removed now

**Review by ilfreddy — APPROVED** [review]

File: python/mrchem/api.py [hunk 6]
```diff
@@ -127,7 +127,8 @@ def write_molecule(user_dict, origin):
         "charge": mol.charge,
         "coords": mol.get_coords_in_program_syntax(),
     }
-    if user_dict["WaveFunction"]["environment"].lower() == "pcm":
+    
+    if (user_dict["WaveFunction"]["environment"].lower()).split("_")[0] == "pcm":
```
Comment (robertodr) [inline python/mrchem/api.py:131]: wouldn't this work as well?
```suggestion
    if "pcm" in user_dict["WaveFunction"]["environment"].lower():
```

Comment (Gabrielgerez) [reply python/mrchem/api.py:131]: I think I tried something similar, we can add it just to see if it still works

File: python/mrchem/validators.py [hunk 7]
```diff
@@ -140,7 +140,7 @@ def __init__(self, user_dict, origin):
             self.atomic_coords = self.ang2bohr_array(self.atomic_coords)
 
         # Cavity related data
-        if user_dict["WaveFunction"]["environment"].lower().split("-")[0] == "pcm":
+        if user_dict["WaveFunction"]["environment"].lower().split("_")[0] == "pcm":
```
[the remark this answers is not in the document]
Comment (Gabrielgerez) [reply python/mrchem/validators.py]: Changed it in the newest commit

File: python/mrchem/api.py [hunk 8]
```diff
@@ -25,16 +25,11 @@
...
-    write_scf_plot,
-    write_rsp_calc,
-    parse_wf_method,
-)
-from .periodictable import PeriodicTable as PT, PeriodicTableByZ as PT_Z
+from .helpers import (parse_wf_method, write_rsp_calc, write_scf_fock,
+                      write_scf_guess, write_scf_plot, write_scf_properties,
+                      write_scf_solver)
+from .periodictable import PeriodicTable as PT
+from .periodictable import PeriodicTableByZ as PT_Z
```
[the remark this answers is not in the document]
Comment (Gabrielgerez) [reply python/mrchem/api.py:32]: I think, I did not edit this part of the file as far as I remember

**Review by ilfreddy — APPROVED** [review]

**Pull request merged**